src/e2e-test/features/gcsdelete/GCSDelete_WithMacro.feature

```diff
@@ -0,0 +1,46 @@
+@GCSDelete_Action
+Feature: GCS Delete - Verification of GCS Delete plugin with macro arguments
+
+  @GCS_CSV_TEST
+  Scenario: Verify GCS Delete successfully deletes all objects in gcs bucket when bucket name is passed as macro
+    Given Open Datafusion Project to configure pipeline
+    When Expand Plugin group in the LHS plugins list: "Conditions and Actions"
+    When Select plugin: "GCS Delete" from the plugins list as: "Actions"
+    When Navigate to the properties page of plugin: "GCS Delete"
+    Then Enter the GCS Delete property "projectId" as macro argument "gcsDeleteProjectId"
+    Then Enter the GCS Delete property "objectsToDelete" as macro argument "gcsDeleteObjectsToDelete"
+    Then Enter the GCS Delete property "serviceAccountFilePath" as macro argument "gcsServiceAccount"
+    Then Validate "GCS Delete" plugin properties
+    Then Close the GCS Delete properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "projectId" for key "gcsDeleteProjectId"
+    Then Enter runtime argument value bucket name for GCS Delete property objects to delete key "gcsDeleteObjectsToDelete"
+    Then Enter runtime argument value "serviceAccountAutoDetect" for key "gcsServiceAccount"
+    Then Run the Pipeline in Runtime with runtime arguments
+    Then Wait till pipeline is in running state
+    Then Open and capture logs
+    Then Verify the pipeline status is "Succeeded"
+    Then Verify all the objects in the GCS bucket deleted successfully by GCS Delete action plugin
+
+  @GCS_READ_RECURSIVE_TEST
+  Scenario: Verify GCS Delete successfully deletes all objects in gcs bucket when comma separated object list is passed as macro
+    Given Open Datafusion Project to configure pipeline
+    When Expand Plugin group in the LHS plugins list: "Conditions and Actions"
+    When Select plugin: "GCS Delete" from the plugins list as: "Actions"
+    When Navigate to the properties page of plugin: "GCS Delete"
+    Then Enter the GCS Delete property "projectId" as macro argument "gcsDeleteProjectId"
+    Then Enter the GCS Delete property "objectsToDelete" as macro argument "gcsDeleteObjectsToDelete"
+    Then Enter the GCS Delete property "serviceAccountFilePath" as macro argument "gcsServiceAccount"
+    Then Validate "GCS Delete" plugin properties
+    Then Close the GCS Delete properties
+    Then Save and Deploy Pipeline
+    Then Run the Pipeline in Runtime
+    Then Enter runtime argument value "projectId" for key "gcsDeleteProjectId"
+    Then Enter runtime argument value "gcsDeleteObjectsList" as comma separated objects for GCS Delete property objects to delete key "gcsDeleteObjectsToDelete"
```

Comment:
To keep the step definition simple refactor it as 
`Enter runtime argument value "gcsDeleteObjectsList" for key "gcsDeleteObjectsToDelete"`

Reply:
Implementation of this step is specific to GCS delete plugin, to handle comma separated list of objects so the step description should be descriptive.

Reply:
Still the step grammar is unnecessarily big, it can be reduced to something like
`Enter runtime argument value "gcsDeleteObjectsList" as comma separated objects key "gcsDeleteObjectsToDelete"`

Reply:
Change Done.